Code/GraphMol/Chirality.cpp

```diff
@@ -1379,9 +1382,8 @@ void iterateCIPRanks(const ROMol &mol, const DOUBLE_VECT &invars,
     // now truncate each vector and stick the rank at the end
     if (static_cast<unsigned int>(lastNumRanks) != numRanks) {
       for (unsigned int i = 0; i < numAtoms; ++i) {
-        cipEntries[i][numIts + 1] = ranks[i];
-        cipEntries[i].erase(cipEntries[i].begin() + numIts + 2,
-                            cipEntries[i].end());
+        cipEntries[i][cipRankIndex] = ranks[i];
+        cipEntries[i].resize(cipRankIndex + 1);
```

Comment:
With these changes it looks like you can remove the section of code that pads the rounds to be the same length (around line 1360) as well as get rid of the `longestEntry` tracking.

Reply:
Good catch, done!